CMakeLists.txt

```diff
@@ -431,8 +431,31 @@ endif()
 
 if(DEFINED BUILD_VERSION)
   set(build_version_argument "-DBUILD_VERSION=${BUILD_VERSION}")
-elseif(EXISTS ${ZEPHYR_BASE}/.git/index)
+elseif(IS_DIRECTORY ${ZEPHYR_BASE}/.git)
```

Comment:
there is no reason to change this to `IS_DIRECTORY`.

If `.git` is a file, then `if(EXISTS ${ZEPHYR_BASE}/.git/index)`  will be false.
And as we are adding a dependency to `${ZEPHYR_BASE}/.git/index` then that should also be what we check against.

Reply:
Fair enough. It's now back to the explicit test for `.git/index`.